appmap/test/test_recording.py

```diff
@@ -0,0 +1,64 @@
+"""Test recording context manager"""
+import json
+import os
+import platform
+import sys
+from importlib import reload
+
+import pytest
+
+FIXTURE_DIR = os.path.join(
+    os.path.dirname(os.path.realpath(__file__)),
+    'data'
+    )
+
+
+@pytest.mark.datafiles(
+    os.path.join(FIXTURE_DIR, 'appmap.yml'),
+    os.path.join(FIXTURE_DIR, 'example_class.py'),
+    os.path.join(FIXTURE_DIR, 'expected.appmap.json')
+    )
+def test_recording_works(monkeypatch, datafiles):
+    with open(datafiles / 'expected.appmap.json') as f:
+        expected_appmap = json.load(f)
+
+    # Setting these outside the definition of expected_appmap makes it
+    # easier to update when the expected appmap changes
+    py_impl = platform.python_implementation()
+    py_version = platform.python_version()
+    expected_appmap['metadata']['language']['engine'] = py_impl
+    expected_appmap['metadata']['language']['version'] = py_version
+
+
+    sys.path.append(str(datafiles))
+
+    monkeypatch.setenv("APPMAP", "true")
+    monkeypatch.setenv("APPMAP_CONFIG", str(datafiles / 'appmap.yml'))
+    monkeypatch.setenv("APPMAP_LOG_LEVEL", "debug")
+
+    import appmap
+    reload(appmap._implementation.recording)  # pylint: disable=protected-access
```

Comment:
why?

Reply:
Reloading `recording` was necessary to reinitialize it to use the environment variable settings. In thinking about how to answer your question, I realized I didn't really like having most of the initialization done as a side-effect of loading the package.

So, I create an `initialize` function for several of the modules, and called each explicitly. Those changes are in e025a0d.